core/src/actors/blocks_manager/mod.rs

```diff
@@ -154,6 +155,30 @@ impl BlocksManager {
             |block| Ok(block.clone()),
         )
     }
+
+    fn discard_existing_inv_vectors(&mut self, inv_vectors: &[InvVector]) -> InvVectorsResult {
+        // Missing inventory vectors
+        let mut missing_inv_vectors = Vec::new();
+
+        // Iterate through all inventory vectors
+        for inv_vector in inv_vectors {
+            // Get hash from inventory vector
+            let hash = match inv_vector {
+                InvVector::Error(hash)
+                | InvVector::Block(hash)
+                | InvVector::Tx(hash)
+                | InvVector::DataRequest(hash)
+                | InvVector::DataResult(hash) => hash,
+            };
+
+            // Add the inventory vector to the missing vectors if it is not found
+            if self.blocks.get(&hash).is_none() {
+                missing_inv_vectors.push(inv_vector.clone());
```

Comment:
This function would be more clear if written in functional style:

```rust
let missing_inv_vectors = inv_vectors
    .iter()
    .filter(|inv_vector| {
        let hash = /**/
        self.blocks.get(&hash).is_none()
    })
    .collect()
```

Reply:
Also, it may make sense to take ownership of the `inv_vectors`, which would allow to use `into_iter`:
```rust
fn discard_existing_inv_vectors(&mut self, inv_vectors: Vec<InvVector>) -> InvVectorsResult {
```